sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -66,6 +66,10 @@ import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types._
 import org.apache.spark.util.{CircularBuffer, Utils}
 
+private[hive] class CatalogAndHiveTableImpl(
```

Comment:
does this really help? We need to cast an `Object` to `HiveTable` anyway.

Reply:
It just doesn't let us create an instance with something else than `HiveTable`, but not much help indeed.